capella_console_client/search.py

```diff
@@ -149,6 +150,11 @@ def __init__(self, session: CapellaConsoleSession, **kwargs) -> None:
         if sortby:
             self.payload["sortby"] = self._get_sort_payload(sortby)
 
+        input_ownership = kwargs.pop("ownership", None)
+        ownership_payload = self._get_ownership_payload(input_ownership)
+        if ownership_payload:
```

Comment:
actually looking at this again I would restructure this to something like

```python
ownership_option = cur_kwargs.pop("ownership", None)

if ownership_option and self._is_valid_ownership(ownership_option):
    self.payload["ownership"] = ownership_option

...

def _is_valid_ownership(self, ownership_option: str) -> bool:
    is_valid = ownership_option in list(OwnershipOption)
    if not is_valid:
        logger.warning(f"ownership option {ownership_option} not supported - supported values {[o.value for o in OwnershipOption]} ... omitting")
    return is_valid
```


I guess one could (rightfully) argue why `_is_valid_ownership` should be within `StacSearch` and not e.g. the actual enum (or an validator object of that, which I consider overkill)

```python

class OwnershipOption(str, BaseEnum):
    ORG = "ownedByOrganization"
    PUBLIC = "publiclyAvailable"
    SHARED = "sharedWithOrganization"
    PURCHASABLE = "availableForPurchase"

    @classmethod
    def is_valid(cls, option_str: str) -> bool:
        return option_str in list(cls)

```

that way you could drop the client code down to something like

```python
ownership_option_str = cur_kwargs.pop("ownership", None)

if OwnershipOption.is_valid(ownership_option_str):
    self.payload["ownership"] = ownership_option
```

let me know what you think :)